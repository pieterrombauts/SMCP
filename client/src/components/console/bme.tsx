import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import bme_1 from "./../../media/Bme_1.png"
import bme_2 from "./../../media/Bme_2.png"

interface AppProps {};

const bme: React.FC<AppProps> = () => {
  return(
         <Tabs defaultActiveKey="Spartan_console" id="uncontrolled-tab-example"  >
        
          <Tab eventKey="BME_1" title="Astronaut Health">
            <img src={bme_1} style={{width:"300px"}}/> 
          </Tab>

          <Tab eventKey="BME_2" title="Astronaut Vitals">
          <img src={bme_2} style={{width:"300px"}}/> 

          </Tab>
        </Tabs>
  );
}

export const Bme = bme;