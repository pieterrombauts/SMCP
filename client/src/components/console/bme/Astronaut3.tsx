import React from 'react';
import { bmeVitalSignsDisplay } from '../customTypes'

interface A3Props {
  values: bmeVitalSignsDisplay;
};

const Astronaut3: React.FC<A3Props> = ( props ) => {
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="" y="55%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="22%" y="59.5%" width={150} height={150} href='/media/Bme_3.gif'></image>
      <image x="2%" y="52%" width={250} height={500} href='/media/Bme_Christina_Koch.jpg'></image>

      <text x="2%" y="60%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Christina Koch</text>
      <text x="2%" y="95%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 3</text>

      <text x="22%" y="59.5%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="34%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="41%" y="63.5%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="34%" y="66.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut3.heartRate}</text>
      <text x="41%" y="66.5%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut3.heartRateMax}</text>

      <text x="22%" y="76.5%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="22%" y="79.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">115 / 70</text>

      <text x="22%" y="84.5%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
    
    </g>
  )
};

export default Astronaut3;