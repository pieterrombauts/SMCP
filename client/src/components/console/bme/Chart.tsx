import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { Line } from 'react-chartjs-2';
import {chartData} from '../customTypes'

interface ChartProps {
  className?: string;
  data: {x: string, y: number};
};

const UChart: React.FC<ChartProps> = (props) => {
  const [chartData, setChartData] = useState<chartData>({
    datasets: [
      {
        data: [
          {x: '2019-11-15T14:55:00', y: 37.2}
        ],
        backgroundColor: 'white',
        borderColor: 'red',
        borderWidth: 2,
        fill: false,
      }
    ]
  })
  useEffect(() => {
    setChartData({
      datasets: [
        {
          data: [...chartData.datasets[0].data, props.data],
          backgroundColor: 'white',
          borderColor: 'red',
          borderWidth: 2,
          fill: false
        }
      ]
    })
  }, [props.data])
  return (
    <div className={props.className}>
      <Line
        data={chartData}
        width={200}
        height={90}
        redraw={false}
        options={{
          animation: {
            duration: 0
          },
          title: {
            display: false
          },
          legend: {
            display: false
          },
          elements: {
            point: {
              radius: 0
            },
            line: {
              tension: 0
            }
          },
          layout: {
            padding: {
              bottom: 5
            }
          },
          scales: {
            xAxes: [{
              gridLines: {
                color: '#378545',
                lineWidth: 1
              },
              type: 'time',
              ticks: {
                min: '2019-11-15T15:00:00',
                max: '2019-11-15T16:45:00'
              },
              time: {
                unit: 'minute',
                stepSize: 30
              }
            }],
            yAxes: [{
              gridLines: {
                color: '#378545',
                lineWidth: 1
              },
              ticks: {
                min: 35.8,
                max: 37.8,
                stepSize: 0.2
              }
            }]
          },
          bezierCurve: false,
          mainAspectRation: false,
          responsive: true
        }}
      />
    </div>
  )
};

export const Chart = styled(UChart)`
  background-color: black;
  padding: 5px;
`;