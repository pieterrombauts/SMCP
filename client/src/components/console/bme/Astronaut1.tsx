import React, {useState, useEffect} from 'react';
import { bmeVitalSignsDisplay, bmeVitalSignsTemps } from '../customTypes'
import {Chart} from './Chart';
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';

interface A1Props {
  values: bmeVitalSignsDisplay;
  temps: bmeVitalSignsTemps;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time
})

const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & A1Props

const Astronaut1: React.FC<Props> = ( props ) => {
  const [newData, setNewData] = useState<{x: string, y: number}>({x:"2019-11-15T15:00:00", y:37})
  useEffect(() => {
    if (props.time !== "") setNewData({x: props.time, y: props.temps.astronaut1.temperature})
  }, [props.temps])
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect x="0%" y="3%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="22%" y="7.5%" width={150} height={150} href='/media/Bme_1.gif'></image>
      <image x="2%" y="0%" width={250} height={500} href='/media/Bme_Luca_Parmitano.jpg'></image>
      
      <text x="2%" y="8%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Luca Parmitano</text>
      <text x="2%" y="43%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Commander</text>

      <text x="22%" y="8%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="34%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="41%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="34%" y="15%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.heartRate}</text>
      <text x="41%" y="15%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.heartRateMax}</text>

      <text x="22%" y="25%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="22%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.bloodPressureSystolic.toFixed(0)} / </text>
      <text x="26.5%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut1.bloodPressureDiastolic.toFixed(0)}</text>

      <text x="22%" y="33%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <foreignObject x="22%" y="34%" height="125px" width="300px" fill="black"><Chart data={newData}/></foreignObject>
    </g>
  )
};

export default connector(Astronaut1);