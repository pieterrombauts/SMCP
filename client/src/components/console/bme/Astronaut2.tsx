import React, {useState, useEffect} from 'react';
import { bmeVitalSignsDisplay, bmeVitalSignsTemps} from '../customTypes'
import {Chart} from './Chart';
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';

interface A2Props {
  values: bmeVitalSignsDisplay;
  temps: bmeVitalSignsTemps;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time
})

const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & A2Props

export const Astronaut2: React.FC<Props> = ( props ) => {
  const [newData, setNewData] = useState<{x: string, y: number}>({x:"2019-11-15T15:00:00", y:37})
  useEffect(() => {
    if (props.time !== "") setNewData({x: props.time, y: props.temps.astronaut2.temperature})
  }, [props.temps])
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="50%" y="3%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="72%" y="7.5%" width={150} height={150} href='/media/Bme_2.gif'></image>
      <image x="52%" y="0%" width={250} height={500} href='/media/Bme_Andrew_R_Morgan.jpg'></image>
      
      <text x="52%" y="8%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Andrew Morgan</text>
      <text x="52%" y="43%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 2</text>

      <text x="72%" y="8%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="84%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="91%" y="12%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="84%" y="15%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.heartRate}</text>
      <text x="91%" y="15%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.heartRateMax}</text>

      <text x="72%" y="25%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="72%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.bloodPressureSystolic.toFixed(0)} /</text>
      <text x="76.5%" y="28%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut2.bloodPressureDiastolic.toFixed(0)}</text>

      <text x="72%" y="33%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <foreignObject x="72%" y="34%" height="125px" width="300px" fill="black"><Chart data={newData}/></foreignObject>
    </g>
  )
};

export default connector(Astronaut2);