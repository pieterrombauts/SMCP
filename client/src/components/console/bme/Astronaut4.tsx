import React, { useEffect, useState } from 'react';
import { bmeVitalSignsDisplay, bmeVitalSignsTemps } from '../customTypes'
import {Chart} from './Chart';
import { RootState } from '../../../reducers';
import { connect, ConnectedProps } from 'react-redux';

interface A4Props {
  values: bmeVitalSignsDisplay;
  temps: bmeVitalSignsTemps;
};

const mapState = (state: RootState ) => ({
  time: state.dataReducer.time
})

const connector = connect(mapState, {})
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux & A4Props

export const Astronaut4: React.FC<Props> = ( props ) => {
  const [newData, setNewData] = useState<{x: string, y: number}>({x:"2019-11-15T15:00:00", y:37})
  useEffect(() => {
    if (props.time !== "") setNewData({x: props.time, y: props.temps.astronaut4.temperature})
  }, [props.temps])
  return(
    <g id="bme-vital-signs-astronaut1">
      <rect  x="50%" y="51%" rx="5" ry="5" width="650" height="450" style={{fill:"#333333", stroke:"#393939", strokeWidth:5, opacity:0.5}} />
      <image x="72%" y="55.5%" width={150} height={150} href='/media/Bme_4.gif'></image>
      <image x="52%" y="48%" width={250} height={500} href='/media/Bme_Jessica_Meir.jpg'></image>

      <text x="52%" y="56%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Jessica Meir</text>
      <text x="52%" y="91%" fill="#EBB249" fontSize="20" fontFamily="Verdana">Flight Engineer 5</text>
      
      <text x="72%" y="55.5%" fill="white" fontSize="20" fontFamily="Verdana">Heart Rate (BPM)</text>
      <text x="84%" y="59.5%" fill="white" fontSize="12" fontFamily="Verdana">Current BPM</text>
      <text x="91%" y="59.5%" fill="white" fontSize="12" fontFamily="Verdana">Max BPM</text>
      <text x="84%" y="62.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">{props.values.astronaut4.heartRate}</text>
      <text x="91%" y="62.5%" fill="#f74848" fontSize="20" fontFamily="Verdana">{props.values.astronaut4.heartRateMax}</text>

      <text x="72%" y="72.5%" fill="white" fontSize="20" fontFamily="Verdana">Blood Pressure (mmHg)</text>
      <text x="72%" y="75.5%" fill="#7ef56e" fontSize="20" fontFamily="Verdana">115 / 70 </text>

      <text x="72%" y="80.5%" fill="white" fontSize="20" fontFamily="Verdana">Temperature (Â°C)</text>
      <foreignObject x="72%" y="81.5%" height="125px" width="300px" fill="black"><Chart data={newData}/></foreignObject>
    </g>
  )
};

export default connector(Astronaut4);