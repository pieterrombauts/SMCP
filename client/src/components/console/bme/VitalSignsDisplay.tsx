import React, { useEffect, useState, useRef } from 'react';
import styled from 'styled-components';
import Astronaut1 from './Astronaut1';
import Astronaut2 from './Astronaut2'
import Astronaut3 from './Astronaut3'
import Astronaut4 from './Astronaut4'
import { generateRandVal } from '../../../dataGenerator';
import SmoothieComponent, { TimeSeries } from 'react-smoothie';
import { decreaseSystolic } from '../../../bloodPressureSystolic';
import { increaseDiastolic } from '../../../bloodPressureDiastolic';
import { Chart } from './Chart';
// import bme_1 from "./../../media/Bme_1.png"
// import bme_2 from "./../../media/Bme_2.png"

interface VSDProps {
    className?: string;
};

const ts1 = new TimeSeries({});
const ts2 = new TimeSeries({});
const ts3 = new TimeSeries({});
const ts4 = new TimeSeries({});

const defaultValues = {
    astronaut1: {
        heartRate: 0,
        heartRateMax: 0,
        bloodPressureSystolic: 125,
        bloodPressureDiastolic: 60,
        temperature: 0
    },

    astronaut2: {
        heartRate: 0,
        heartRateMax: 0,
        bloodPressureSystolic: 125,
        bloodPressureDiastolic: 60,
        temperature: 0
    },

    astronaut3: {
        heartRate: 0,
        heartRateMax: 0,
        temperature: 0
    },

    astronaut4: {
        heartRate: 0,
        heartRateMax: 0,
        temperature: 0
    }
}


const UVitalSignsDisplay: React.FC<VSDProps> = (props) => {
    var [values, setValues] = useState(defaultValues)
    const intervalRef = useRef(0);

    useEffect(() => {
        if (intervalRef.current !== 0) {
            clearInterval(intervalRef.current);
        }
        const id = setInterval(() => {
            var newValues = {
                astronaut1: {
                    heartRate: generateRandVal(60, 90).toFixed(0),
                    heartRateMax: 91,
                    bloodPressureSystolic: parseFloat(decreaseSystolic(values.astronaut1.bloodPressureSystolic).toFixed(2)),
                    bloodPressureDiastolic: parseFloat(increaseDiastolic(values.astronaut1.bloodPressureDiastolic).toFixed(2)),
                    temperature: generateRandVal(36.5, 37.5).toFixed(1)
                },

                astronaut2: {
                    heartRate: generateRandVal(60, 90).toFixed(0),
                    heartRateMax: 91,
                    bloodPressureSystolic: parseFloat(decreaseSystolic(values.astronaut2.bloodPressureSystolic).toFixed(2)),
                    bloodPressureDiastolic: parseFloat(increaseDiastolic(values.astronaut2.bloodPressureDiastolic).toFixed(2)),
                    temperature: generateRandVal(36.5, 37.5).toFixed(1)
                },

                astronaut3: {
                    heartRate: generateRandVal(60, 80).toFixed(0),
                    heartRateMax: 91,
                    temperature: generateRandVal(36.5, 37.5).toFixed(1)
                },

                astronaut4: {
                    heartRate: generateRandVal(60, 80).toFixed(0),
                    heartRateMax: 91,
                    temperature: generateRandVal(36.5, 37.5).toFixed(1)
                },

            }
            setValues(values = newValues);

            setInterval(function() {
                var time = new Date().getTime();
               
                ts1.append(time, values.astronaut1.temperature);
                ts2.append(time, values.astronaut2.temperature);
                ts3.append(time, values.astronaut3.temperature);
                ts4.append(time, values.astronaut4.temperature);
              }, 5000);

        }, 5000);
        intervalRef.current = id;
    }, [])

    return (

        <div className={props.className}>
            <div id="chart1-div">
                <Chart />
            </div>

            <div id="chart2-div">
                <Chart />
            </div>

            <div id="chart3-div">
                <Chart />
            </div>

            <div id="chart4-div">
                <Chart />
            </div>
           
            <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1400 1000'>
                <Astronaut1 values={values} />
                <Astronaut2 values={values} />
                <Astronaut3 values={values} />
                <Astronaut4 values={values} />
            </svg>
        </div>
    );
}

export const VitalSignsDisplay = styled(UVitalSignsDisplay)`
  width: 100%;
  height: 100%;
  position: absolute;

  #chart1-div{ 
    position: absolute;
    height: 125px;
    width: 3000px;
    top: 34%;
    left: 22%;
    background-color: black;

  }

  #chart1-div{ 
    position: absolute;
    height: 125px;
    width: 300px;
    top: 34%;
    left: 22%;
    background-color: black;

  }
  

  #chart2-div{ 
    position: absolute;
    height: 125px;
    width: 300px;
    background-color: black;
    top: 34%;
    left: 72%;
    

  }

  #chart3-div{ 
    position: absolute;
    height: 125px;
    width: 300px;
    
    background-color: black;
    top: 81.5%;
    left: 22%;

  }

  #chart4-div{ 
    position: absolute;
    height: 125px;
    width: 300px;
    background-color: black;
    top: 81.5%;
    left: 72%;

  }

`;