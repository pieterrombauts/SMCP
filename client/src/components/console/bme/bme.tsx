import React, { useState } from 'react';
import styled from 'styled-components';
import Nav from 'react-bootstrap/Nav';
import Tab from 'react-bootstrap/Tab';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { VitalSignsDisplay } from './VitalSignsDisplay'
import { EVASuitDisplay } from './EVASuitDisplay'
import socket from '../../Socket'

interface AppProps {
  className?: string;
};

const UBme: React.FC<AppProps> = (props) => {
  const [ firstOpensVS, setFirstOpensVS ] = useState<boolean>(true);
  function updateFirstConsoleVSOpen(event: React.MouseEvent<any>) {
    if (firstOpensVS) {
      socket.emit('FIRST_CONSOLE_OPEN', 'bme-vs')
      setFirstOpensVS(false);
    }
  }
  return (
    <Tab.Container id="bme-tabs" defaultActiveKey="EVASuit">
      <Row className={props.className}>
        <Col id="bme-content" sm={10}>
          <Tab.Content>
            <Tab.Pane eventKey="EVASuit">
              <EVASuitDisplay />
            </Tab.Pane>
            <Tab.Pane onClick={updateFirstConsoleVSOpen} eventKey="vitalSigns">
              <VitalSignsDisplay />
            </Tab.Pane>
          </Tab.Content>
        </Col>
        <Col id="bme-buttons" sm={2}>
          <Nav variant="pills" className="flex-column">
            <Nav.Item>
              <Nav.Link eventKey="EVASuit">EVA Suit</Nav.Link>
            </Nav.Item>
            <Nav.Item>
              <Nav.Link onClick={updateFirstConsoleVSOpen} eventKey="vitalSigns">Vital Signs</Nav.Link>
            </Nav.Item>
          </Nav>
        </Col>
      </Row>
    </Tab.Container>

  );
}

export const Bme = styled(UBme)`
  width: 100%;
  height: 100%;
  position: absolute;

  #bme-buttons {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  #bme-buttons .nav {
    margin-left: 20px;
    max-width: 300px;
    background-color: #f8f9fa;
    border-radius: 3px;
  }

  #bme-buttons .nav-item a {
    padding: 20px 30px;
  }
`;