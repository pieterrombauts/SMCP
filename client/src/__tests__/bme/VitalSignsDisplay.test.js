import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { UVitalSignsDisplay } from '../../components/console/bme/VitalSignsDisplay'
import { shallow, mount, render} from 'enzyme';

describe("Vital signs display", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<UVitalSignsDisplay />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})