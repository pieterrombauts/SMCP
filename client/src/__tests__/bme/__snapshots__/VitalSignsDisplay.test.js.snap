// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vital signs display renders correctly 1`] = `
<div>
  <svg
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 1400 1000"
  >
    <Connect(Astronaut1)
      temps={
        Object {
          "astronaut1": Object {
            "temperature": 0,
          },
          "astronaut2": Object {
            "temperature": 0,
          },
          "astronaut3": Object {
            "temperature": 0,
          },
          "astronaut4": Object {
            "temperature": 0,
          },
        }
      }
      values={
        Object {
          "astronaut1": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut2": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut3": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut4": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
        }
      }
    />
    <Connect(Astronaut2)
      temps={
        Object {
          "astronaut1": Object {
            "temperature": 0,
          },
          "astronaut2": Object {
            "temperature": 0,
          },
          "astronaut3": Object {
            "temperature": 0,
          },
          "astronaut4": Object {
            "temperature": 0,
          },
        }
      }
      values={
        Object {
          "astronaut1": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut2": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut3": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut4": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
        }
      }
    />
    <Connect(Astronaut3)
      temps={
        Object {
          "astronaut1": Object {
            "temperature": 0,
          },
          "astronaut2": Object {
            "temperature": 0,
          },
          "astronaut3": Object {
            "temperature": 0,
          },
          "astronaut4": Object {
            "temperature": 0,
          },
        }
      }
      values={
        Object {
          "astronaut1": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut2": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut3": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut4": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
        }
      }
    />
    <Connect(Astronaut4)
      temps={
        Object {
          "astronaut1": Object {
            "temperature": 0,
          },
          "astronaut2": Object {
            "temperature": 0,
          },
          "astronaut3": Object {
            "temperature": 0,
          },
          "astronaut4": Object {
            "temperature": 0,
          },
        }
      }
      values={
        Object {
          "astronaut1": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut2": Object {
            "bloodPressureDiastolic": 60,
            "bloodPressureSystolic": 125,
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut3": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
          "astronaut4": Object {
            "heartRate": 0,
            "heartRateMax": 0,
          },
        }
      }
    />
  </svg>
</div>
`;
